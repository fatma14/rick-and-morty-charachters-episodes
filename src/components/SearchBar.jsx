import React from "react";

export default function SearchBar({ searchPhrase, handleChange }) {
  return (
    <div>
      <label>
        <input
          type="text"
          value={searchPhrase}
          onChange={(event) => {
            event.preventDefault();
            handleChange(event.target.value);
          }}
        />
      </label>
    </div>
  );
}
