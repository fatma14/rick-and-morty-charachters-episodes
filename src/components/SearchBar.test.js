import React from "react";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";

describe("SearchBar unit tests", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <SearchBar searchPhrase="ein" handleChange={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call handle change", () => {
    const handleChange = jest.fn();
    const value = "test";
    const component = renderer.create(
      <SearchBar searchPhrase="kal" handleChange={handleChange} />
    );
    component.root
      .findByType("input")
      .props.onChange({ preventDefault: () => {}, target: { value: value } });

    expect(handleChange).toBeCalledWith(value);
  });
});
