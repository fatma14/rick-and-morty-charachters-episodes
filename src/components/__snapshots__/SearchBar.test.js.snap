// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar unit tests should match snapshot 1`] = `
<div
  className="searchBarContainer"
>
  <label>
    <input
      className="searchBar"
      onChange={[Function]}
      placeholder="Search character..."
      type="text"
      value="ein"
    />
  </label>
</div>
`;
